hive-metastore/src/main/java/org/apache/iceberg/hive/HiveClientPool.java

```diff
@@ -47,10 +49,15 @@ public HiveClientPool(int poolSize, Configuration conf) {
   }
 
   @Override
-  protected HiveMetaStoreClient newClient()  {
+  protected boolean shouldRetry() {
+    return false; // Already use RetryingMetaStoreClient
+  }
+
+  @Override
+  protected IMetaStoreClient newClient()  {
     try {
       try {
-        return CLIENT_CTOR.newInstance(hiveConf);
+        return CLIENT_CTOR.invoke(hiveConf, true);
```

Comment:
Can you please add a comment that documents the `true` here? Something like this:

```
  return GET_CLIENT.invoke(hiveConf, true /* use retries */);
```

I'm assuming that the `true` is retry related? If so, should this detect whether the Hive implementation will retry and turn retries back on?

Reply:
it's not about retry, but to configure `allowEmbedded` for the `HiveMetaStoreClient` initialized internally in the `RetryingMetaStoreClient`:

https://github.com/apache/hive/blob/9f03e3c416834b18faf7a85338b130ed0742275f/standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java#L165-L167

Reply:
Yes, thanks @jackye1995  for the pointer.  From quick glance, seems a flag just intended to determine whether to throw an exception if "hive.metastore.uris" is not set, so not terribly useful to document, unless you feel otherwise.